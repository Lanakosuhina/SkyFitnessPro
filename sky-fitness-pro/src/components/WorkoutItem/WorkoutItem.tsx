import React from "react";

type WorkoutItemType = {
  id: string;
  isDone: boolean;
  workoutName: string;
  setSelected: React.Dispatch<React.SetStateAction<string>>;
};

export default function WorkoutItem({
  id,
  isDone,
  workoutName,
  setSelected,
}: WorkoutItemType) {
  return (
    <li className="relative  h-16 border-solid border-b border-selectionBorder py-[10px] ml-1 mr-[26px] flex flex-col justify-center">
        <input
          type="radio"
          className="hidden"
          name="selectWorkout"
          id={`workout${id}`}
          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
            setSelected(e.target.value)
          }
          value={id}
        />
        <label
          htmlFor={`workout${id}`}
          className="p-3 pl-[34px] blok rounded-[30px]  font-roboto-400 text-[18px] md:text-[24px]  leading-[110%]"
        >
          {workoutName}
          {isDone && (
          <svg
            className="absolute top-[22px] left-2"
            width="20"
            height="20"
            viewBox="0 0 20 20"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fill-rule="evenodd"
              clip-rule="evenodd"
              d="M10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20ZM9.91339 14.1459L15.4134 7.64594L13.8866 6.35406L9.1373 11.9669L6.40258 8.8415L4.89742 10.1585L8.39742 14.1585C8.58922 14.3777 8.86704 14.5024 9.15829 14.5C9.44953 14.4976 9.72525 14.3683 9.91339 14.1459Z"
              fill="#BCEC30"
            />
          </svg>
        )}
          <svg
            className="absolute top-[22px] left-2"
            width="20.000000"
            height="20.000000"
            viewBox="0 0 20 20"
            fill="none"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <circle
              id="Ellipse 3845"
              cx="10.000000"
              cy="10.000000"
              r="9.500000"
              stroke="#000000"
              strokeOpacity="1"
              strokeWidth="1.000000"
            />
          </svg>
        </label>
    </li>
  );
}
