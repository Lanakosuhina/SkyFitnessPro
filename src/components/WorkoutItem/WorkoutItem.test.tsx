import renderer from "react-test-renderer";
import WorkoutItem from './WorkoutItem'

describe('Компонент input type=radio', () => {
  const mocksetSelected = jest.fn();

  it('корректно рендериться с начальными значениями', () => {
    const component = renderer.create(
      <WorkoutItem
      id={'dfhjk8'}
      isDone={false}
      workoutName={'Тренировка мышц ягодиц'}
      setSelected={mocksetSelected}
      />).toJSON();
    expect(component).toMatchSnapshot();
  });

});

