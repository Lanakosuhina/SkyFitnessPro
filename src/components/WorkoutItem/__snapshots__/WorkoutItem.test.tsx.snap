// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент input type=radio корректно рендериться с начальными значениями 1`] = `
<li
  className="relative  h-16 border-solid border-b border-selectionBorder py-[10px] ml-1 mr-[26px] flex flex-col justify-center"
>
  <input
    className="hidden"
    id="workoutdfhjk8"
    name="selectWorkout"
    onChange={[Function]}
    type="radio"
    value="dfhjk8"
  />
  <label
    className="p-3 pl-[34px] blok rounded-[30px]  font-roboto-400 text-[18px] md:text-[24px]  leading-[110%]"
    htmlFor="workoutdfhjk8"
  >
    Тренировка мышц ягодиц
    <svg
      className="absolute top-[22px] left-2"
      fill="none"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20ZM9.91339 14.1459L15.4134 7.64594L13.8866 6.35406L9.1373 11.9669L6.40258 8.8415L4.89742 10.1585L8.39742 14.1585C8.58922 14.3777 8.86704 14.5024 9.15829 14.5C9.44953 14.4976 9.72525 14.3683 9.91339 14.1459Z"
        fill="#D0CECE"
        fillRule="evenodd"
      />
    </svg>
  </label>
</li>
`;
